Samples/AudioPreprocessing/WinUI/AudioPreprocessing/AudioPreprocessing/Model/PreprocessingModel.cs

```diff
@@ -0,0 +1,244 @@
+                .SetOutput("reshaped", "reshaped_output"))
+              .Operators.Add(new Operator("MatMul")
+                .SetInput("A", "reshaped_output")
+                .SetInput("B", "mel_weight_matrix")
+                .SetOutput("Y", "mel_spectrogram"))
+              .Operators.Add(new Operator("Reshape")
+                .SetInput("data", "mel_spectrogram")
+                .SetConstant("shape", TensorInt64Bit.CreateFromArray(new List<long>() { 4 }, melSpectrogramShape))
+                .SetOutput("reshaped", "Output.MelSpectrogram"));
+
+            var model = builder.CreateModel();
+
+            LearningModelSession session = new LearningModelSession(model);
+            LearningModelBinding binding = new LearningModelBinding(session);
+
+            binding.Bind("Input.TimeSignal", TensorFloat.CreateFromArray(signalShape, signal.ToArray()));
+
+            // Bind output
+            var outputImage = new VideoFrame(
+                BitmapPixelFormat.Bgra8,
+                nMelBins,
+                nDfts);
+
+
+            binding.Bind("Output.MelSpectrogram", outputImage);
+
+            // Evaluate
+            var sw = Stopwatch.StartNew();
+            var result = session.Evaluate(binding, "");
+            sw.Stop();
+            Console.WriteLine("Evaluate Took: %f\n", sw.ElapsedMilliseconds);
+
+            return outputImage.SoftwareBitmap;
+        }
+
+        static void ModelBuilding_MelWeightMatrix()
+        {
+            long[] outputShape = { 9, 8 };
+            var builder =
+                LearningModelBuilder.Create(13)
+                .Outputs.Add(LearningModelBuilder.CreateTensorFeatureDescriptor("Output.MelWeightMatrix", TensorKind.Float, outputShape))
+                .Operators.Add(new Operator("MelWeightMatrix", MicrosoftExperimentalDomain)
+                    .SetConstant("num_mel_bins", TensorInt64Bit.CreateFromArray(new List<long>(), new long[] { 8 }))
+                    .SetConstant("dft_length", TensorInt64Bit.CreateFromArray(new List<long>(), new long[] { 16 }))
+                    .SetConstant("sample_rate", TensorInt64Bit.CreateFromArray(new List<long>(), new long[] { 8192 }))
+                    .SetConstant("lower_edge_hertz", TensorFloat.CreateFromArray(new List<long>(), new float[] { 0 }))
+                    .SetConstant("upper_edge_hertz", TensorFloat.CreateFromArray(new List<long>(), new float[] { (float)(8192 / 2.0) }))
+                    .SetOutput("output", "Output.MelWeightMatrix"));
+
+            var model = builder.CreateModel();
+            LearningModelSession session = new LearningModelSession(model);
+            LearningModelBinding binding = new LearningModelBinding(session);
+            var result = session.Evaluate(binding, "");
+
+            Console.WriteLine("\n");
+            Console.WriteLine("Output.MelWeightMatrix\n");
+        }
+
+        static void STFT(int batchSize, int signalSize, int dftSize,
```

Comment:
can you either rename the method or put a comment describing what "STFT" is?

Reply:
It's been removed! I thought this method was adding the operator to the domain. But it was just a test method I copied over from the LearningModel sample that Sheil had.

Fwiw a Short Time Fourier Transform (STFT) takes a short moving window over a long audio file and calculates the fourier transform over the small snippets. Mel Spectrograms are often calculated on the fourier transforms of these small windows.